client/blocks/subscription-length-picker/index.jsx

```diff
@@ -0,0 +1,193 @@
+import { TERM_MONTHLY } from 'lib/plans/constants';
+import SubscriptionLengthOption from './option';
+
+export class SubscriptionLengthPicker extends React.Component {
+	static propTypes = {
+		initialValue: PropTypes.string,
+		currencyCode: PropTypes.string.isRequired,
+		productsWithPrices: PropTypes.array.isRequired,
+		onChange: PropTypes.func.isRequired,
+		translate: PropTypes.func.isRequired,
+	};
+
+	static defaultProps = {
+		onChange: () => null,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			checked: props.initialValue,
+		};
+	}
+
+	render() {
+		const { productsWithPrices, translate } = this.props;
+		return (
+			<div className="subscription-length-picker">
+				<div className="subscription-length-picker__header">
+					{ translate( 'Choose the length of your subscription', {
+						comment:
+							'We offer a way to change billing term from default bill every 1 year to something ' +
+							'else like bill once every 2 years - this is related header.',
+					} ) }
+				</div>
+
+				<div className="subscription-length-picker__options">
+					{ productsWithPrices.map( p => this.renderProduct( p ) ) }
+				</div>
+			</div>
+		);
+	}
+
+	renderProduct = p => {
+		const { plan, planSlug } = p;
+
+		return (
+			<div className="subscription-length-picker__option-container" key={ planSlug }>
+				<SubscriptionLengthOption
+					term={ plan.term }
+					checked={ planSlug === this.state.checked }
+					price={ myFormatCurrency( p.priceFull, this.props.currencyCode ) }
+					pricePerMonth={ myFormatCurrency( p.priceMonthly, this.props.currencyCode ) }
+					savePercent={ Math.round( 100 * ( 1 - p.priceMonthly / this.getHighestMonthlyPrice() ) ) }
+					value={ planSlug }
+					onCheck={ this.handleCheck }
+				/>
+			</div>
+		);
+	};
+
+	getHighestMonthlyPrice() {
+		return Math.max( ...this.props.productsWithPrices.map( p => Number( p.priceMonthly ) ) );
+	}
+
+	handleCheck = ( { value } ) => {
+		this.setState( {
+			checked: value,
+		} );
+		this.props.onChange( { value } );
+	};
+}
+
+class SubscriptionLengthPickerContainer extends React.Component {
+	static defaultProps = {
+		...SubscriptionLengthPicker.defaultProps,
+
+		fetchingPlans: PropTypes.bool.isRequired,
+		fetchingProducts: PropTypes.bool.isRequired,
+		requestProductsList: PropTypes.func.isRequired,
+		requestPlans: PropTypes.func.isRequired,
+	};
+
+	componentWillMount() {
+		if ( this.props.productsWithPrices.length === 0 ) {
+			if ( ! this.props.fetchingProducts ) {
+				this.props.requestProductsList();
+			}
+			if ( ! this.props.fetchingPlans ) {
+				this.props.requestPlans();
+			}
+		}
+	}
+
+	render() {
+		return <SubscriptionLengthPicker { ...this.props } />;
+	}
+}
+
+const EPSILON = 0.009;
+
+function myFormatCurrency( price, code ) {
+	const hasCents = Math.abs( price % 1 ) > EPSILON;
+	return formatCurrency( price, code, {
+		precision: hasCents ? 2 : 0,
+	} );
+}
+
+const computeProductsWithPrices = ( state, plans ) => {
```

Comment:
There is a ton going on here. :) 

1. As an overall thought, maybe we need to update API endpoints/redux state to better meet our UI needs.
2. As a first step can we compose what's happening into smaller testable functions?
3. Perhaps those functions would be better suited as new state selectors?
4. computeProductsWithPrices is called in `mapStateWithProps` which is called whenever global state is modified. Selectors/functions we call there need to be speedy. If we create new selectors consider if we should memoize it.
5. Tests are much appreciated!